### PR TITLE
Upgrade to the Firebase 3.x.x SDK

### PR DESCRIPTION
**Pull request by abehaskins**

### CODE REVIEW

**Comment by abehaskins** [comment]
I signed it!

**Comment by abehaskins** [comment]
@jwngr Ptal then lets ship this :fire: :fire: :fire: :fire: :fire: :fire:

**Comment by musdy** [comment]
merge merge merge!

**Comment by matiishyn** [comment]
yes, please merge it

**Comment by katowulf** [comment]
Hey guys, we'll get to this soon. It's 53 files, so it's going to take a bit to review.

There's no rush to get upgraded. The previous SDK versions aren't going anywhere. Hang in there!

**Comment by damien-mcmahon** [comment]
Any update on the status of this?

File: src/utils.js [hunk 1]
```diff
@@ -368,7 +360,12 @@
             var def = utils.defer();
             if( angular.isFunction(ref.set) || !angular.isObject(data) ) {
               // this is not a query, just do a flat set
-              ref.set(data, utils.makeNodeResolver(def));
+              try {
+                ref.set(data, utils.makeNodeResolver(def));
```
Comment (jwngr) [inline src/utils.js]: Should this be a `try` / `catch` or a `then` / `catch`?

File: src/FirebaseArray.js [hunk 2]
```diff
@@ -109,12 +109,25 @@
...
+          var ref = this.$ref().ref.push();
+          var dataJSON;
+
+          try {
+            dataJSON = $firebaseUtils.toJSON(data);
+          } catch (err) {
+            def.reject(err);
+          }
+
+          if (dataJSON) {
```
Comment (jwngr) [inline src/FirebaseArray.js]: This check technically changes behavior unless we can prove that `dataJSON` is never empty. I'm not sure if we actually want the check here. For example, if `dataJSON` is `{}`, we should probably allow that write (which will essentially do a remove).

**Pull request merged**